pkg/provider/subnets.go

```diff
@@ -68,8 +68,8 @@ func (p *SubnetProvider) Get(ctx context.Context, nodeTemplate *v1alpha1.AWSNode
 		return nil, fmt.Errorf("no subnets matched selector %v", nodeTemplate.Spec.SubnetSelector)
 	}
 	p.cache.SetDefault(fmt.Sprint(hash), output.Subnets)
-	subnetLog := prettySubnets(output.Subnets)
-	if p.cm.HasChanged("subnets", subnetLog) {
+	subnetLog := PrettySubnets(output.Subnets)
+	if p.cm.HasChanged(fmt.Sprintf("subnets-ids (%s-%s)", nodeTemplate.Kind, nodeTemplate.Name), subnetLog) {
```

Comment:
Should this be what it's like in the security group code here on on securitygroups.go#L109? This makes more sense to me.